pkg/kapp/clusterapply/add_or_update_change.go

```diff
@@ -69,6 +69,11 @@ func (c AddOrUpdateChange) ApplyStrategy() (ApplyStrategy, error) {
 	case ctldiff.ChangeOpUpdate:
 		newRes := c.change.NewResource()
 
+		_, found := newRes.Annotations()[ctldiff.MaxDurationAnnKey]
+		if found {
+			return UpdateAlwaysReplaceStrategy{c}, nil
```

Comment:
Why should it always pick this strategy implicitly if it has the annotation?

Won't the diff stage pick up the update since the resource is modified in the diff stage?

Reply:
Diff stage will pick the update and update the resource with the update strategy which is already set for that particular resource. For an example, resource has plain update strategy then kapp will do simple update for that resource and in simple update resource's creationTimestamp does not change. Hence our required condition (creationTimestap + max-duration < current time) will always pass after that. 

To change the creationTimestamp on every update we need to delete and create the resource and for that we have to use update strategy as always-replace.

Reply:
Lets depend on the added timestamp as well and not always pick this strategy